### DISCUSSION TITLE
Help Needed: Trouble with passible TableInfo instances in migration functions.

### DISCUSSION THREAD

**Discussion by mwakaambrose** [simolus3/drift#2473] [Q&A] [answered]
I am trying to add a migration to my app. on the doc they has something like 

```dart
@override
MigrationStrategy get migration {
  return MigrationStrategy(
    onCreate: (Migrator m) async {
      await m.createAll();
    },
    onUpgrade: (Migrator m, int from, int to) async {
      if (from == 12) {
          await m.addColumn(table, table.column);
```
And this is the signation of the addColumn function.
`Future<void> addColumn(TableInfo<Table, dynamic> table, GeneratedColumn<Object> column)`

What do I pass as the table?

For a bit more context, I have everything inside a database.dart file. This includes the tables too.

```dart
part 'database.g.dart';

class SomeTable extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get remoteId => text().nullable()();
  TextColumn get name => text()();
  IntColumn get age => int().nullable(); // new column
}
// More tables...

@DriftDatabase(tables: [
  SomeTable,
])
class MyDatabase extends _$MyDatabase {
  MyDatabase() : super(_openConnection());
  @override
  int get schemaVersion => 1;
}
@override
MigrationStrategy get migration {
  return MigrationStrategy(
    onCreate: (Migrator m) async {
      await m.createAll();
    },
    onUpgrade: (Migrator m, int from, int to) async {
      if (from == 1) {
        await m.addColumn(someTable, someTable.age);
      }
    },
  );
}

LazyDatabase _openConnection() {
  return LazyDatabase(() async {
    final dbFolder = await getApplicationDocumentsDirectory();
    final file = File(p.join(dbFolder.path, 'db.sqlite'));
    return NativeDatabase(file, logStatements: false);
  });
}
```
This `await m.addColumn(someTable, someTable.age);` doesn't work. Syntax error and it wont compile and it can't import anything. I know I am missing something small but I just can't figure it out.

Please help!

**Comment by simolus3** [comment, marked as answer]
This works for me:

```dart
class SomeTable extends Table {
  IntColumn get id => integer().autoIncrement()();
  TextColumn get remoteId => text().nullable()();
  TextColumn get name => text()();
  IntColumn get age => integer().nullable()(); // new column
}
// More tables...

@DriftDatabase(tables: [
  SomeTable,
])
class MyDatabase extends _$MyDatabase {
  MyDatabase() : super(_openConnection());
  @override
  int get schemaVersion => 2;

  @override
  MigrationStrategy get migration {
    return MigrationStrategy(
      onCreate: (Migrator m) async {
        await m.createAll();
      },
      onUpgrade: (Migrator m, int from, int to) async {
        if (from == 1) {
          await m.addColumn(someTable, someTable.age);
        }
      },
    );
  }
}
```

I've made the following fixes from your source snippet:

1. The `migration` callback needs to be _inside_ the database class for the `@override` to work - that will also make `someTable` visible.
2. The definition of `age` should be `integer().nullable()(); // new column`. The table from your snippet did not define the column correctly, drift also reports this as an error during the build step.
3. I've also changed the `schemaVersion` getter to return `2`. Otherwise, there's no way for `onUpgrade` to get called for existing installations.

**Reply by mwakaambrose** [reply]
Hi @simolus3 this worked. I mostly missed the fact that the override needs to be inside the database class. Thanks for this pointer. Cheers!
